nodes/object_operators/ScFindNearest.py

```diff
@@ -19,20 +21,45 @@ def init(self, context):
         self.inputs.new("ScNodeSocketNumber", "Distance").init("in_distance")
         self.outputs.new("ScNodeSocketVector", "Location")
         self.outputs.new("ScNodeSocketVector", "Normal")
-        self.outputs.new("ScNodeSocketNumber", "Index")
+        self.outputs.new("ScNodeSocketNumber", "Face Index")
         self.outputs.new("ScNodeSocketNumber", "Distance")
+        self.outputs.new("ScNodeSocketVector", "Vertex Position")
+        self.outputs.new("ScNodeSocketNumber", "Vertex Index")
     
     def error_condition(self):
         return(
             super().error_condition()
             or self.inputs["Distance"].default_value < 0.0
         )
-    
+
     def post_execute(self):
         out = super().post_execute()
-        ret = mathutils.bvhtree.BVHTree().FromObject(self.inputs["Object"].default_value, bpy.context.evaluated_depsgraph_get()).find_nearest(self.inputs["Origin"].default_value, self.inputs["Distance"].default_value)
-        out["Location"] = ret[0]
-        out["Normal"] = ret[1]
-        out["Index"] = ret[2]
-        out["Distance"] = ret[3]
-        return out
\ No newline at end of file
+        ret = self.inputs["Object"].default_value.closest_point_on_mesh(self.inputs["Origin"].default_value, distance=self.inputs["Distance"].default_value)
+        out["Location"] = ret[1]
+        out["Normal"] = ret[2]
+        out["Face Index"] = ret[3]
+        out["Distance"] = self.measure_distance(self.inputs["Origin"].default_value, out["Location"])
+        current_mode = bpy.context.object.mode
+        bpy.ops.object.mode_set(mode='EDIT')
+        bm = bmesh.from_edit_mesh(self.inputs["Object"].default_value.data)
+        bm.faces.ensure_lookup_table()
+        face = bm.faces[out["Face Index"]]
+        closest_vertex = Vector((0,0,0))
+        closest_index = -1
+        closest_distance = 10 ** 10
+        for v in face.verts:
```

Comment:
It could be possible that in some cases the closest vertex isn't a part of a polygon, and is only connected by an edge (or not even that).

Reply:
I honestly missed that. But the reason that I used the face index was two fold. 1. I have it already, and 2. Checking a face should take far less time then checking the whole mesh.

I figured using the already known face would be better because checking all vertices could be a huge performance hit depending on the size of the mesh.

I'll see what I can do to make this more performant and squash the inputs.